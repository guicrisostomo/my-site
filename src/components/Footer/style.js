import styled from "styled-components";

export const FooterStyle = styled.footer`
    background: rgba(24, 26, 27, 0.95);
    color: #e0e0e0;
    box-shadow: 0 -2px 24px 0 rgba(0,0,0,0.18);
    backdrop-filter: blur(6px);
    border-top: 1.5px solid #23272F;
    padding-top: 18px;
    padding-bottom: 8px;
    width: 100%;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    position: relative;
    bottom: 0;
`;

export const ImgCopyRight = styled.img`
    height: 22px;
    width: 22px;
    margin-left: 7px;
    margin-right: 7px;
    align-self: center;
    filter: brightness(0.8);

    @media (max-width: 690px) {
        height: 14px;
        width: 14px;
    }
`;

export const DivTextACreditToMe = styled.div`
    font-size: 18px;
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 10px 0 0 0;
    letter-spacing: 0.5px;
    font-family: 'Fira Mono', 'Roboto Mono', monospace;
    color: #bdbdbd;

    @media (max-width: 690px) {
        font-size: 13px;
    }
`;

export const DivContacts = styled.div`
    display: flex;
    flex-direction: row;
    justify-content: center;
    align-items: center;
    gap: 22px;
    padding: 8px 0 0 0;
`;

export const Icons = styled.a`
    padding: 8px;
    border-radius: 50%;
    margin-left: 0;
    background: linear-gradient(135deg, #23272F 60%, #181A1B 100%);
    color: #00e676;
    height: 40px;
    width: 40px;
    text-align: center;
    font-size: 28px;
    display: flex;
    align-items: center;
    justify-content: center;
    box-shadow: 0 2px 8px rgba(0,0,0,0.12);
    border: 1.5px solid #23272F;
    transition: all 0.25s;

    &:hover {
        color: #fff;
        background: linear-gradient(135deg, #00e676 60%, #00bfa5 100%);
        border-color: #00e676;
        transform: scale(1.08) translateY(-2px);
        box-shadow: 0 4px 16px rgba(0,255,128,0.10);
    }

    @media (max-width: 690px) {
        height: 28px;
        width: 28px;
        font-size: 18px;
        padding: 4px;
    }
`;