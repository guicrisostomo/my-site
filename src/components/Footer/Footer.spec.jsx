import { render, screen } from "@testing-library/react";
import Footer from "./index";
import imgCopyright from "../../assets/img/imgCopyright.svg";

describe("Footer", () => {
  it("should render the footer", () => {
    render(<Footer />);
    const footer = screen.getByTestId("footer");
    expect(footer).toBeInTheDocument();
  });

  it("should render the icons", () => {
    render(<Footer />);
    const icons = screen.getAllByRole("link");
    expect(icons).toHaveLength(3);
  });

  it("should render the text", () => {
    render(<Footer />);
    const text = screen.getByText(/Guilherme Crisostomo da Silva/i);
    expect(text).toBeInTheDocument();
  });

  it("should render the image", () => {
    render(<Footer />);
    const img = screen.getByRole("img");
    expect(img).toBeInTheDocument();
  });

  it("should render the image with the correct src", () => {
    render(<Footer />);
    const img = screen.getByRole("img");
    expect(img.src).toContain(imgCopyright);
  });

  it("should match snapshot", () => {
    const { asFragment } = render(<Footer />);
    expect(asFragment()).toMatchSnapshot();
  });
})