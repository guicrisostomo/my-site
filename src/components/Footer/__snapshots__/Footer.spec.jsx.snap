// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should match snapshot 1`] = `
<DocumentFragment>
  <footer
    class="sc-bczRLJ iyrEsR"
    data-testid="footer"
  >
    <div
      class="sc-hKMtZM fIUnKK"
    >
      <a
        class="sc-eCYdqJ hjYTFe"
        href="https://github.com/guicrisostomo"
        target="_blank"
      >
        <svg
          class="icon-github"
          cursor="pointer"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M511.6 76.3C264.3 76.2 64 276.4 64 523.5 64 718.9 189.3 885 363.8 946c23.5 5.9 19.9-10.8 19.9-22.2v-77.5c-135.7 15.9-141.2-73.9-150.3-88.9C215 726 171.5 718 184.5 703c30.9-15.9 62.4 4 98.9 57.9 26.4 39.1 77.9 32.5 104 26 5.7-23.5 17.9-44.5 34.7-60.8-140.6-25.2-199.2-111-199.2-213 0-49.5 16.3-95 48.3-131.7-20.4-60.5 1.9-112.3 4.9-120 58.1-5.2 118.5 41.6 123.2 45.3 33-8.9 70.7-13.6 112.9-13.6 42.4 0 80.2 4.9 113.5 13.9 11.3-8.6 67.3-48.8 121.3-43.9 2.9 7.7 24.7 58.3 5.5 118 32.4 36.8 48.9 82.7 48.9 132.3 0 102.2-59 188.1-200 212.9a127.5 127.5 0 0 1 38.1 91v112.5c.8 9 0 17.9 15 17.9 177.1-59.7 304.6-227 304.6-424.1 0-247.2-200.4-447.3-447.5-447.3z"
          />
        </svg>
      </a>
      <a
        class="sc-eCYdqJ hjYTFe"
        href="https://www.linkedin.com/in/guilherme-crisostomo-da-silva"
        target="_blank"
      >
        <svg
          cursor="pointer"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M847.7 112H176.3c-35.5 0-64.3 28.8-64.3 64.3v671.4c0 35.5 28.8 64.3 64.3 64.3h671.4c35.5 0 64.3-28.8 64.3-64.3V176.3c0-35.5-28.8-64.3-64.3-64.3zm0 736c-447.8-.1-671.7-.2-671.7-.3.1-447.8.2-671.7.3-671.7 447.8.1 671.7.2 671.7.3-.1 447.8-.2 671.7-.3 671.7zM230.6 411.9h118.7v381.8H230.6zm59.4-52.2c37.9 0 68.8-30.8 68.8-68.8a68.8 68.8 0 1 0-137.6 0c-.1 38 30.7 68.8 68.8 68.8zm252.3 245.1c0-49.8 9.5-98 71.2-98 60.8 0 61.7 56.9 61.7 101.2v185.7h118.6V584.3c0-102.8-22.2-181.9-142.3-181.9-57.7 0-96.4 31.7-112.3 61.7h-1.6v-52.2H423.7v381.8h118.6V604.8z"
          />
        </svg>
      </a>
      <a
        class="sc-eCYdqJ hjYTFe"
        href="mailto:guilherme.silva9872@gmail.com"
        target="_blank"
      >
        <svg
          cursor="pointer"
          fill="currentColor"
          height="1em"
          stroke="currentColor"
          stroke-width="0"
          viewBox="0 0 1024 1024"
          width="1em"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
          />
        </svg>
      </a>
    </div>
    <div
      class="sc-dkzDqf bLetck"
    >
      <img
        class="sc-gsnTZi iZVpCW"
        src="imgCopyright.svg"
      />
      Guilherme Crisostomo da Silva
    </div>
  </footer>
</DocumentFragment>
`;
